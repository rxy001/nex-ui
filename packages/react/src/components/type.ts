export type ClassNames = string | string[]

export type ComponentClasses<OwnerState, T extends string> = Partial<
  Record<T, ClassNames | ((ownerState: OwnerState) => ClassNames)>
>
