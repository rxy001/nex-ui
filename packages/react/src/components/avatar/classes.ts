import { generateUtilityClasses } from '../utils'

export const avatarClasses = generateUtilityClasses('nui-avatar', [
  'root',

  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-xl',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'outlined',

  'img',
])

export const avatarGroupClasses = generateUtilityClasses('nui-avatar-group', [
  'root',
  'surplus',
])
