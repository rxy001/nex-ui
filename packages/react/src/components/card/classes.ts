import { generateUtilityClasses } from '../utils'

export const cardClasses = generateUtilityClasses('nui-card', [
  'root',
  'blurred',
  'hoverable',

  'shadow-xs',
  'shadow-sm',
  'shadow-md',
  'shadow-lg',
  'shadow-xl',

  'radius-none',
  'radius-sm',
  'radius-md',
  'radius-lg',
])

export const cardHeaderClasses = generateUtilityClasses('nui-card-header', [
  'root',
  'content',
  'title',
  'subtitle',
])

export const cardBodyClasses = generateUtilityClasses('nui-card-body', ['root'])

export const cardFooterClasses = generateUtilityClasses('nui-card-footer', [
  'root',
])

export const cardActionAreaClasses = generateUtilityClasses(
  'nui-card-action-area',
  ['root'],
)
