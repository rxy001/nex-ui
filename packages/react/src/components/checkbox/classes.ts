import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const checkboxClasses = generateUtilityClasses('nui-checkbox', [
  // ---------root---------
  'root',

  // "sm" | "md" | "lg" | "full" | "none"
  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  'indeterminate',
  // ---------root---------

  'input',
  'label',
  'icon',
])

export const checkboxGroupClasses = generateUtilityClasses(
  'nui-checkbox-group',
  [
    'root',
    'label',
    'wrapper',
    'orientation-vertical',
    'orientation-horizontal',
  ],
)
