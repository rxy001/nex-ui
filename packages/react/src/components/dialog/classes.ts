import { generateUtilityClasses } from '../utils'

export const dialogClasses = generateUtilityClasses('nui-dialog', [
  'root',
  'open',
  'backdrop',
])

export const dialogContentClasses = generateUtilityClasses(
  'nui-dialog-content',
  [
    'root',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'full-screen',

    'placement-top',
    'placement-bottom',
    'placement-center',

    'scroll-inside',
    'scroll-outside',

    'paper',
    'close-button',
  ],
)

export const dialogHeaderClasses = generateUtilityClasses('nui-dialog-header', [
  'root',
])

export const dialogFooterClasses = generateUtilityClasses('nui-dialog-footer', [
  'root',
])

export const dialogBodyClasses = generateUtilityClasses('nui-dialog-body', [
  'root',
])
