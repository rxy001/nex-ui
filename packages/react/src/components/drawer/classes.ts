import { generateUtilityClasses } from '../utils'

export const drawerClasses = generateUtilityClasses('nui-drawer', [
  'root',
  'open',
  'backdrop',
])

export const drawerContentClasses = generateUtilityClasses(
  'nui-drawer-content',
  [
    'root',

    'placement-top',
    'placement-right',
    'placement-bottom',
    'placement-left',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'paper',
    'close-button',
  ],
)

export const drawerHeaderClasses = generateUtilityClasses('nui-drawer-header', [
  'root',
])

export const drawerFooterClasses = generateUtilityClasses('nui-drawer-footer', [
  'root',
])

export const drawerBodyClasses = generateUtilityClasses('nui-drawer-body', [
  'root',
])
