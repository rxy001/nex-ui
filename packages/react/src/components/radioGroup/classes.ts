import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const radioClasses = generateUtilityClasses('nui-radio', [
  // ---------root---------
  'root',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  // ---------root---------

  'input',
  'label',
  'dot',
])

export const radioGroupClasses = generateUtilityClasses('nui-radio-group', [
  'root',
  'label',
  'wrapper',
  'orientation-vertical',
  'orientation-horizontal',
])
