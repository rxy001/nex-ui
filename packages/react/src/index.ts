import './types/declare'

export * from './components'
export * from './theme'

export type { Theme } from './types/theme'
export type { ComponentsTheme } from './types/componentsTheme'

export { useColorScheme } from '@nex-ui/system'
export { nex, styled } from '@nex-ui/styled'
