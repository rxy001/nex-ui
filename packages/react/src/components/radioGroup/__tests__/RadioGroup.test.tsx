import { createRef, useState } from 'react'
import {
  renderWithNexUIProvider,
  testComponentStability,
  testRootClassName,
} from '~/tests/shared'
import { act, fireEvent } from '@testing-library/react'
import { RadioGroup, Radio } from '../index'
import { radioGroupClasses, radioClasses } from '../classes'
import type { ElementType } from 'react'
import type { RadioGroupProps } from '../index'

describe('RadioGroup', () => {
  function TestComponent(props: RadioGroupProps<string, ElementType>) {
    return (
      <RadioGroup {...props}>
        <Radio value='orange'>Orange</Radio>
        <Radio value='pear'>Pear</Radio>
        <Radio value='apple'>Apple</Radio>
      </RadioGroup>
    )
  }

  testComponentStability(<TestComponent />)

  testRootClassName(<TestComponent className='test-class' />, 'test-class')

  it('should render with default props', () => {
    const { container } = renderWithNexUIProvider(<TestComponent />)
    const root = container.firstElementChild

    expect(root).toHaveClass(radioGroupClasses.root)
    expect(root).toHaveClass(radioGroupClasses['orientation-horizontal'])
    expect(root).not.toHaveClass(radioGroupClasses['orientation-vertical'])

    expect(root).toMatchSnapshot()
  })

  it('should forward classes to root, label and wrapper slots', () => {
    const classes = {
      root: 'test-root',
      label: 'test-label',
      wrapper: 'test-wrapper',
    }

    const { container, queryByClassName } = renderWithNexUIProvider(
      <TestComponent classes={classes} label='Label' />,
    )

    expect(container.firstElementChild).toHaveClass(classes.root)
    expect(queryByClassName(radioGroupClasses.label)).toHaveClass(classes.label)
    expect(queryByClassName(radioGroupClasses.wrapper)).toHaveClass(
      classes.wrapper,
    )
  })

  it('should forward slotProps to label and wrapper slots', () => {
    const slotProps = {
      label: {
        className: 'test-label',
      },
      wrapper: {
        className: 'test-wrapper',
      },
    }

    const { queryByClassName } = renderWithNexUIProvider(
      <TestComponent slotProps={slotProps} label='Label' />,
    )

    expect(queryByClassName(radioGroupClasses.label)).toHaveClass(
      slotProps.label.className,
    )
    expect(queryByClassName(radioGroupClasses.wrapper)).toHaveClass(
      slotProps.wrapper.className,
    )
  })

  it("should forward ref to RadioGroup's root element", () => {
    const ref = createRef<HTMLDivElement>()
    const { getByTestId } = renderWithNexUIProvider(
      <TestComponent data-testid='radiogroup' ref={ref} />,
    )

    expect(ref.current).toBe(getByTestId('radiogroup'))
  })

  it('should render with orientation class based on orientation prop', () => {
    const { getByTestId } = renderWithNexUIProvider(
      <>
        <TestComponent orientation='vertical' data-testid='vertical' />
        <TestComponent orientation='horizontal' data-testid='horizontal' />
      </>,
    )

    expect(getByTestId('vertical')).toHaveClass(
      radioGroupClasses['orientation-vertical'],
    )
    expect(getByTestId('horizontal')).toHaveClass(
      radioGroupClasses['orientation-horizontal'],
    )
  })

  it('should switch different values when radios are clicked', () => {
    const { getAllByRole } = renderWithNexUIProvider(<TestComponent />)

    const [orangeRadio, pearRadio, appleRadio] = getAllByRole('radio')

    expect(orangeRadio).not.toBeChecked()
    expect(pearRadio).not.toBeChecked()
    expect(appleRadio).not.toBeChecked()

    fireEvent.click(orangeRadio)
    expect(orangeRadio).toBeChecked()

    fireEvent.click(pearRadio)
    expect(pearRadio).toBeChecked()

    fireEvent.click(appleRadio)
    expect(appleRadio).toBeChecked()
  })

  it('should be controlled by value prop', () => {
    function ControlledRadioGroup() {
      const [value, setValue] = useState('orange')

      return (
        <>
          <TestComponent value={value} onValueChange={setValue} />
          <div data-testid='current-value'>{value}</div>
        </>
      )
    }

    const { getAllByRole, getByTestId } = renderWithNexUIProvider(
      <ControlledRadioGroup />,
    )

    const [orangeRadio, pearRadio, appleRadio] = getAllByRole('radio')
    const currentValue = getByTestId('current-value')

    expect(orangeRadio).toBeChecked()
    expect(pearRadio).not.toBeChecked()
    expect(appleRadio).not.toBeChecked()
    expect(currentValue).toHaveTextContent('orange')

    fireEvent.click(pearRadio)
    expect(pearRadio).toBeChecked()
    expect(orangeRadio).not.toBeChecked()
    expect(appleRadio).not.toBeChecked()
    expect(currentValue).toHaveTextContent('pear')

    fireEvent.click(appleRadio)
    expect(appleRadio).toBeChecked()
    expect(orangeRadio).not.toBeChecked()
    expect(pearRadio).not.toBeChecked()
    expect(currentValue).toHaveTextContent('apple')
  })

  it('should support defaultValue prop', () => {
    const { getAllByRole } = renderWithNexUIProvider(
      <TestComponent defaultValue='apple' />,
    )

    const [orangeRadio, pearRadio, appleRadio] = getAllByRole('radio')

    expect(orangeRadio).not.toBeChecked()
    expect(pearRadio).not.toBeChecked()
    expect(appleRadio).toBeChecked()
  })

  it('should disable all radios when disabled prop is true', () => {
    const { getAllByRole } = renderWithNexUIProvider(<TestComponent disabled />)

    const radios = getAllByRole('radio')

    radios.forEach((radio) => {
      expect(radio).toBeDisabled()
    })
  })

  it('should apply color and size props to radios', () => {
    const { getAllByRole } = renderWithNexUIProvider(
      <TestComponent color='green' size='lg' />,
    )

    const radios = getAllByRole('radio')

    radios.forEach((radio) => {
      expect(radio.parentElement).toHaveClass(radioClasses['color-green'])
      expect(radio.parentElement).toHaveClass(radioClasses['size-lg'])
    })
  })

  it('should do nothing when a radio without a value is clicked', async () => {
    const { getAllByRole } = renderWithNexUIProvider(
      <RadioGroup>
        <Radio>Orange</Radio>
        <Radio value='pear'>Pear</Radio>
      </RadioGroup>,
    )

    const [firstRadio, secondRadio] = getAllByRole('radio')

    await act(async () => {
      fireEvent.click(firstRadio)
    })
    expect(firstRadio).not.toBeChecked()

    await act(async () => {
      fireEvent.click(secondRadio)
    })
    expect(secondRadio).toBeChecked()
  })

  it('should check non-input elements within RadioGroup when clicked', async () => {
    const { getByRole } = renderWithNexUIProvider(
      <RadioGroup>
        <Radio value='test' as='div'>
          Test Radio
        </Radio>
      </RadioGroup>,
    )
    const radio = getByRole('radio')
    const radioRoot = radio.parentElement!

    expect(radioRoot).not.toHaveClass(radioClasses.checked)
    expect(radio).not.toBeChecked()
    await act(async () => {
      fireEvent.click(radio)
    })
    expect(radioRoot).toHaveClass(radioClasses.checked)
    expect(radio).toBeChecked()
  })

  describe('Accessibility', () => {
    it('should have role="radiogroup" by default', () => {
      const { getByTestId } = renderWithNexUIProvider(
        <TestComponent data-testid='radiogroup' />,
      )
      expect(getByTestId('radiogroup')).toHaveAttribute('role', 'radiogroup')
    })

    it('should apply aria-labelledby to the root element when label is provided', () => {
      const { getByTestId, getByRole, rerender } = renderWithNexUIProvider(
        <TestComponent data-testid='radiogroup' />,
      )

      const radioGroup = getByTestId('radiogroup')
      expect(radioGroup).not.toHaveAttribute('aria-labelledby')

      rerender(<TestComponent data-testid='radiogroup' label='Label' />)

      const heading = getByRole('heading', { name: 'Label' })
      expect(radioGroup).toHaveAttribute('aria-labelledby', heading.id)
    })

    it('should move focus to the first radio when Tab is pressed and no radio is checked', async () => {
      const { getAllByRole, user } = renderWithNexUIProvider(<TestComponent />)

      const [firstRadio, secondRadio, thirdRadio] = getAllByRole('radio')

      await user.tab()
      expect(document.activeElement).toBe(firstRadio)

      expect(firstRadio).toHaveAttribute('tabindex', '0')
      expect(secondRadio).toHaveAttribute('tabindex', '-1')
      expect(thirdRadio).toHaveAttribute('tabindex', '-1')
    })

    it('should move focus to the checked radio when Tab is pressed and a radio is checked', async () => {
      const { getAllByRole, user } = renderWithNexUIProvider(
        <TestComponent value='apple' />,
      )

      const [firstRadio, secondRadio, thirdRadio] = getAllByRole('radio')

      await user.tab()
      expect(document.activeElement).toBe(thirdRadio)

      expect(firstRadio).toHaveAttribute('tabindex', '-1')
      expect(secondRadio).toHaveAttribute('tabindex', '-1')
      expect(thirdRadio).toHaveAttribute('tabindex', '0')
    })

    it('should move focus to the non-disabled radio when Tab is pressed', async () => {
      const { getAllByRole, user } = renderWithNexUIProvider(
        <RadioGroup>
          <Radio value='orange' disabled>
            Orange
          </Radio>
          <Radio value='pear'>Pear</Radio>
          <Radio value='apple'>Apple</Radio>
        </RadioGroup>,
      )
      const [, secondRadio] = getAllByRole('radio')

      await user.tab()
      expect(document.activeElement).toBe(secondRadio)
    })

    it('should move focus to the correct radio when Tab is pressed and multiple RadioGroups exist', async () => {
      const { user } = renderWithNexUIProvider(
        <>
          <TestComponent value='apple' />
          <TestComponent value='pear' />
          <TestComponent value='orange' />
        </>,
      )

      await user.tab()
      expect(document.activeElement).toHaveAttribute('value', 'apple')
      await user.tab()
      expect(document.activeElement).toHaveAttribute('value', 'pear')
      await user.tab()
      expect(document.activeElement).toHaveAttribute('value', 'orange')
    })

    it('should move focus to and check the next radio when ➡️ or ⬇️ is pressed', async () => {
      const { getAllByRole, user } = renderWithNexUIProvider(<TestComponent />)

      const [firstRadio, secondRadio, thirdRadio] = getAllByRole('radio')

      await user.tab()
      await user.keyboard('{ArrowDown}')
      expect(document.activeElement).toBe(secondRadio)
      expect(secondRadio).toBeChecked()

      await user.keyboard('{ArrowRight}')
      expect(document.activeElement).toBe(thirdRadio)
      expect(thirdRadio).toBeChecked()

      await user.keyboard('{ArrowDown}')
      expect(document.activeElement).toBe(firstRadio)
      expect(firstRadio).toBeChecked()
    })

    it('should move focus to and check the previous radio when ⬅️ or ⬆️ is pressed', async () => {
      const { getAllByRole, user } = renderWithNexUIProvider(<TestComponent />)

      const [firstRadio, secondRadio, thirdRadio] = getAllByRole('radio')

      await user.tab()
      await user.keyboard('{ArrowLeft}')
      expect(document.activeElement).toBe(thirdRadio)
      expect(thirdRadio).toBeChecked()

      await user.keyboard('{ArrowUp}')
      expect(document.activeElement).toBe(secondRadio)
      expect(secondRadio).toBeChecked()

      await user.keyboard('{ArrowLeft}')
      expect(document.activeElement).toBe(firstRadio)
      expect(firstRadio).toBeChecked()
    })

    it('should move focus to and check the non-disabled radio', async () => {
      const { getAllByRole, user, rerender } = renderWithNexUIProvider(
        <RadioGroup>
          <Radio value='orange'>Orange</Radio>
          <Radio value='pear' disabled>
            Pear
          </Radio>
          <Radio value='apple'>Apple</Radio>
        </RadioGroup>,
      )
      const [firstRadio, , thirdRadio] = getAllByRole('radio')

      await user.tab()
      expect(document.activeElement).toBe(firstRadio)
      await user.keyboard('{ArrowRight}')
      expect(document.activeElement).toBe(thirdRadio)
      expect(thirdRadio).toBeChecked()

      await user.keyboard('{ArrowLeft}')
      expect(document.activeElement).toBe(firstRadio)
      expect(firstRadio).toBeChecked()

      await user.keyboard('{ArrowDown}')
      expect(document.activeElement).toBe(thirdRadio)
      expect(thirdRadio).toBeChecked()

      await user.keyboard('{ArrowUp}')
      expect(document.activeElement).toBe(firstRadio)
      expect(firstRadio).toBeChecked()

      rerender(
        <RadioGroup>
          <Radio value='orange'>Orange</Radio>
          <Radio value='pear' disabled>
            Pear
          </Radio>
          <Radio value='apple' disabled>
            Apple
          </Radio>
        </RadioGroup>,
      )

      expect(document.activeElement).toBe(firstRadio)
      await user.keyboard('{ArrowUp}')
      expect(document.activeElement).toBe(firstRadio)
      await user.keyboard('{ArrowDown}')
      expect(document.activeElement).toBe(firstRadio)
      await user.keyboard('{ArrowLeft}')
      expect(document.activeElement).toBe(firstRadio)
      await user.keyboard('{ArrowRight}')
      expect(document.activeElement).toBe(firstRadio)
    })
  })
})
