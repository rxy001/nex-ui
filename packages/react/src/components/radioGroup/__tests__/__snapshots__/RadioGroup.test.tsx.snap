// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render with default props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--nui-spaces-2);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  padding: var(--nui-spaces-1-5);
  height: var(--nui-sizes-8);
  -webkit-margin-start: calc(var(--nui-spaces-1-5) * -1);
  margin-inline-start: calc(var(--nui-spaces-1-5) * -1);
  -webkit-margin-end: var(--nui-spaces-1-5);
  margin-inline-end: var(--nui-spaces-1-5);
}

.emotion-3 {
  all: unset;
  position: absolute;
  inset: 0;
  opacity: 0;
  z-index: 1;
}

.emotion-3:is(:hover, [data-hover=true]):not(:disabled, [data-disabled=true])+* {
  border-color: var(--nui-colors-blue-secondary);
}

.emotion-3:is(:focus-visible, [data-focus-visible=true]):not(:disabled, [data-disabled=true])+* {
  outline: var(--nui-borders-focus-visible-outline);
  outline-offset: var(--nui-spaces-0-5);
}

.emotion-4 {
  border-radius: var(--nui-radii-full);
  border: var(--nui-borders-md);
  border-color: var(--nui-colors-gray-highlight);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  box-sizing: border-box;
  -webkit-transition: var(--nui-transitions-colors);
  transition: var(--nui-transitions-colors);
  width: var(--nui-sizes-5);
  height: var(--nui-sizes-5);
}

.emotion-4::before {
  content: "";
  display: block;
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  border-radius: var(--nui-radii-full);
  background-color: var(--nui-colors-blue-primary);
  -webkit-transition: var(--nui-transitions-all);
  transition: var(--nui-transitions-all);
  width: var(--nui-sizes-2);
  height: var(--nui-sizes-2);
}

.emotion-5 {
  -webkit-margin-start: var(--nui-spaces-2);
  margin-inline-start: var(--nui-spaces-2);
  font-size: var(--nui-font-sizes-lg);
}

<div
  class="nui-radio-group-root nui-radio-group-orientation-horizontal emotion-0"
  role="radiogroup"
>
  <div
    class="nui-radio-group-wrapper emotion-1"
  >
    <label
      class="nui-radio-root nui-radio-size-md nui-radio-color-blue emotion-2"
    >
      <input
        aria-label="Orange"
        aria-labelledby="«rc»"
        class="nui-radio-input emotion-3"
        name="«ra»"
        tabindex="0"
        type="radio"
        value="orange"
      />
      <span
        class="nui-radio-dot emotion-4"
      />
      <span
        class="nui-radio-label emotion-5"
        id="«rc»"
      >
        Orange
      </span>
    </label>
    <label
      class="nui-radio-root nui-radio-size-md nui-radio-color-blue emotion-2"
    >
      <input
        aria-label="Pear"
        aria-labelledby="«rd»"
        class="nui-radio-input emotion-3"
        name="«ra»"
        tabindex="-1"
        type="radio"
        value="pear"
      />
      <span
        class="nui-radio-dot emotion-4"
      />
      <span
        class="nui-radio-label emotion-5"
        id="«rd»"
      >
        Pear
      </span>
    </label>
    <label
      class="nui-radio-root nui-radio-size-md nui-radio-color-blue emotion-2"
    >
      <input
        aria-label="Apple"
        aria-labelledby="«re»"
        class="nui-radio-input emotion-3"
        name="«ra»"
        tabindex="-1"
        type="radio"
        value="apple"
      />
      <span
        class="nui-radio-dot emotion-4"
      />
      <span
        class="nui-radio-label emotion-5"
        id="«re»"
      >
        Apple
      </span>
    </label>
  </div>
</div>
`;
